docs/utils/masked/index.rst

```diff
@@ -0,0 +1,165 @@
+.. |Masked| replace:: :class:`~astropy.utils.masked.Masked`
+.. |Quantity| replace:: :class:`~astropy.units.Quantity`
+.. _utils-masked:
+
+**************************************
+Masked Values (`astropy.utils.masked`)
+**************************************
+
+Often, data sets are incomplete or corrupted and it would be handy to be able
+to mask certain values.  Astropy provides a |Masked| class to help represent
+such data sets.
+
+.. warning:: |Masked| is experimental! While we hope basic usage will remain
+   similar, we are not yet sure whether it will not be necessary to change it
+   to make things work throughout Astropy. This also means that comments and
+   suggestions for improvements are especially welcome!
+
+.. note:: |Masked| is similar to Numpy's :class:`~numpy.ma.MaskedArray`,
+   but it supports subclasses much better and also has some important
+   :ref:`differences in behaviour <utils-masked-vs-numpy-maskedarray>`.
+
+Usage
+=====
+
+Astropy |Masked| instances behave like `~numpy.ndarray` or subclasses such as
+|Quantity| but with a mask associated, which is propagated in operations such
+as addition, etc.::
+
+  >>> import numpy as np
+  >>> from astropy import units as u
+  >>> from astropy.utils.masked import Masked
+  >>> ma = Masked([1., 2., 3.], mask=[False, False, True])
+  >>> ma
+  MaskedNDArray([1., 2., ——])
+  >>> mq = ma * u.m
+  >>> mq + 25 * u.cm
+  <MaskedQuantity [1.25, 2.25,  ———] m>
+
+You can get the values without the mask using
+`~astropy.utils.masked.Masked.unmasked`, or, if you need to control what
+should be substituted for any masked values, with
+:meth:`~astropy.utils.masked.Masked.unmask`::
+
+  >>> mq.unmasked
+  <Quantity [1., 2., 3.] m>
+  >>> mq.unmask(fill_value=-75*u.cm)
```

Comment:
Again, I would vote to call this `mq.filled(fill_value=-75*u.cm)`. This way users / developers don't need to know if a column is a MaskedColumn or a MaskedQuantity in order to fill it.

That same argument applies to the `data` attribute. Existing code which assumes the data under the mask in an astropy Table masked column is in `data` will just work with `MaskedQuantity`.

Reply:
This one I don't feel as strongly about - it is mostly that `unmasked = property(unmask)` just seems nice. But `.filled()` tells the reader more clearly what is happening if you give the value as a positional argument. I'm also less worried about name clashes.

Reply:
p.s. And in the current implementation it isn't even true anymore that `unmasked=property(unmask)`...

Still, thinking ahead to a `SkyCoord`, is `.filled()` logical unless you are already thinking that it is a masked class? (The idea would be that the method would always exist, and if no mask is present, would just return the instance itself; I think `sc.unmask(fill_value=...)` would be clearer.

But then again, I've had quite a few times when writing tests that I wrote `...filled()`...

Reply:
OK, since I even called all my tests "filled", let's just go with the familiar! Done in the latest push.